server/service/handler.go

```diff
@@ -500,18 +503,53 @@ func (h *errorHandler) Handle(ctx context.Context, err error) {
 	}
 }
 
+const (
+	CtxStartTimeKey = "start_time"
+)
+
+func markRequestStart(ctx context.Context, r *http.Request) context.Context {
+	return context.WithValue(ctx, CtxStartTimeKey, time.Now())
+}
```

Comment:
Please follow the existing pattern of defining a new package for contexts (eg. https://github.com/fleetdm/fleet/blob/main/server/contexts/authz/authz.go).